From cd225a10f33e0393c5d6a3613eec1a8176cf2dfc Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 15 Feb 2022 19:31:01 +0200
Subject: [PATCH] [SW-75428] gaudi3: perform prstn initialization

This patch add support to perform prstn initialization for both
die0 and die1. prstn initialization includes unmask of prstn
mask and reset PSOC and ARC mask as part of PRSTN reset.

Change-Id: I6c2dc25cfdce1094670833e41a7dd2541f3229b6
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 2ccefa1..9d6f5bc 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -72,6 +72,23 @@
 	return 0;
 }
 
+static void arcboot_rom_prstn_init(void)
+{
+	if (arcboot_rom_global.die_id) {
+		regmap_writel(0, mmD1_PSOC_GLOBAL_CONF_PRSTN_MASK);
+		regmap_writel(D1_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG_EN_M,
+				mmD1_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
+		regmap_writel(D1_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG_EN_M,
+				mmD1_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
+	} else {
+		regmap_writel(0, mmD0_PSOC_GLOBAL_CONF_PRSTN_MASK);
+		regmap_writel(D0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG_EN_M,
+				mmD0_PSOC_RESET_CONF_PSOC_PRSTN_RST_CFG);
+		regmap_writel(D0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG_EN_M,
+				mmD0_PSOC_RESET_CONF_ARC_PRSTN_RST_CFG);
+	}
+}
+
 static void arcboot_rom_read_cold_flops(void)
 {
 	struct gaudi3_cold_rst_data cold_rst_data;
@@ -131,6 +148,8 @@
 
 	arcboot_rom_status_update(PROGRESS_INIT);
 
+	arcboot_rom_prstn_init();
+
 	arcboot_rom_read_cold_flops();
 
 	/* TODO: check whether required for die1 */
